src/authenticator.rs

```diff
@@ -539,7 +539,18 @@ where
 
             let maybe_credential: Option<CredentialFlat> = match file {
                 None => None,
-                Some(c) => self.state.decrypt_content(&mut self.trussed, c).ok(),
+                Some(c) => {
+                    let (res, k) = self
+                        .state
+                        .decrypt_content::<_, CredentialFlat>(&mut self.trussed, c);
+                    if let Ok(mut cred) = res {
+                        cred.encryption_key_type = k;
+                        Some(cred)
+                    } else {
+                        warn_now!("Failed decryption for file {:?}", file_index);
+                        None
+                    }
+                }
```

Comment:
Maybe move this into a helper function like `CredentialFlat::load`?

Reply:
Makes sense - I think I have duplicated this logic anyway

Reply:
Refactored this a bit, please check again